lib/dry/configurable/instance_methods.rb

```diff
@@ -33,9 +33,7 @@ module InstanceMethods
       # Finalize the config and freeze the object
       #
       # @api public
-      def finalize!
-        return self if frozen?
```

Comment:
`#frozen?` checks would be unreliable now and I don't think that it's performance-critical part, just drop them.

Reply:
@ojab I wonder if this should just be idempotent or maybe if we should raise an error if a config is finalized and something tries to finalize it again 🤔

Reply:
Oh my, missed this comment. We can raise an error, but now it's idempotent, so didn't changed it.
IMHO raising an error only would lead to people figuring how to workaround it and will not catch any real issues.